src/Components/DrillPageMinimal.js

```diff
@@ -0,0 +1,159 @@
+import React, { useLayoutEffect } from 'react';
+import { ScrollView, StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
+import { connect } from 'react-redux';
+
+import Progress from './ProgressBar2';
+import DrillAnimation from './DrillAnimation';
+
+import iconCheckMark from '../Images/ic_check.png';
+
+import theme from '../styles/theme.style';
+
+const mapStateToProps = state => {
+  return {
+    drills: state.drills,
+  };
+};
+
+export const DrillPageMinimal = props => {
+  const { route, navigation, drills } = props;
+  const { drill, training } = route.params;
+
+  const totalDrills = training.drills.length;
+  const currentDrill = training.drills.findIndex(id => id === drill.id);
+
+  if (currentDrill === -1) navigation.navigate('TrainingPage', { training });
+
+  const onProgressDotPress = idx => {
+    const drill = drills.find(drill => drill.id === training.drills[idx]);
+    navigation.navigate('DrillPageMinimal', { drill, training });
+  };
+
+  const goToFullDrill = () => {
+    navigation.navigate('DrillPage', { drill });
+  };
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: () => {
+        return (
+          <TouchableOpacity
```

Comment:
There is a HeaderButton inside shared folder that does exactly the same thing.

You could use it here (and tweak it if you want it to use the iconCheckMark image)